OpenOversight/app/utils.py

```diff
@@ -348,20 +348,29 @@ def filter_by_form(form_data, officer_query, department_id=None):
             .all()
         ]
 
-        print(form_data)
         if "Not Sure" in form_data["rank"]:
             form_data["rank"].append(None)
 
-    if form_data.get("badge") or form_data.get("unit") or job_ids:
+    unit_ids = []
+    if form_data.get("unit"):
+        unit_ids = [
+            unit.id
+            for unit in Unit.query.filter_by(department_id=department_id)
+            .filter(Unit.descrip.in_(form_data.get("unit")))
+            .all()
+        ]
+
+        if "Not Sure" in form_data["unit"]:
+            form_data["unit"].append(None)
```

Comment:
How does this work exactly? This might warrant a comment to explain why appending `None` is necessary.

Reply:
Great call, can do!